pandas/tests/frame/test_apply.py

```diff
@@ -718,12 +718,51 @@ def apply_list(row):
 
     def test_apply_noreduction_tzaware_object(self):
         # https://github.com/pandas-dev/pandas/issues/31505
-        df = pd.DataFrame({"foo": [pd.Timestamp("2020", tz="UTC")]}, dtype="object")
+        df = pd.DataFrame(
+            {"foo": [pd.Timestamp("2020", tz="UTC")]}, dtype="datetime64[ns, UTC]"
+        )
         result = df.apply(lambda x: x)
         tm.assert_frame_equal(result, df)
         result = df.apply(lambda x: x.copy())
         tm.assert_frame_equal(result, df)
 
+    def test_apply_function_runs_once(self):
+        # https://github.com/pandas-dev/pandas/issues/30815
+
+        df = pd.DataFrame({"a": [1, 2, 3]})
+        names = []  # Save row names function is applied to
+
+        def reducing_function(row):
+            names.append(row.name)
+
+        def non_reducing_function(row):
+            names.append(row.name)
+            return row
+
+        for func in [reducing_function, non_reducing_function]:
+            del names[:]
+
+            df.apply(func, axis=1)
+            assert names == list(df.index)
+
+    def test_applymap_function_runs_once(self):
+
```

Comment:
is this a specific issue number? you have 3 listed issues can you make sure that we have tests that cover all

Reply:
I Only found a task to remove the documentation about the bug (even though it was unsolved) https://github.com/pandas-dev/pandas/pull/31695
I will search again.
Should i open an issue if there is no issue regarding this?

Reply:
Couldn't find an issue for that except for this old one https://github.com/pandas-dev/pandas/issues/25878

Should i open an issue and add to the test + solve it with this PR?